dev.zed.Zed.yaml

```diff
@@ -0,0 +1,105 @@
+# dev.zed.Zed.yaml
+app-id: dev.zed.Zed
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: zed
+separate-locales: false
+finish-args:
+  # Grant access to Direct Rendering Infrastructure (DRI) for GPU acceleration
+  - --device=dri
+
+  # Allow access to the user's home directory for reading and writing files
+  - --filesystem=home
+
+  # Allow shared memory access for inter-process communication
+  - --share=ipc
+
+  # Enable network access
+  - --share=network
+
+  # Provide access to X11 display server as a fallback
+  - --socket=fallback-x11
+
+  # Enable access to PulseAudio sound server
+  - --socket=pulseaudio
+
+  # Enable access to Wayland display server
+  - --socket=wayland
+
+  # Enable access to Canonical AppMenu
+  - --talk-name=com.canonical.AppMenu.Registrar
+  - --talk-name=com.canonical.AppMenu.Registrar.*
+
+  # Enable access to D-Bus session bus (--socket=session-bus is not supported by Flathub's build system)
+  - --system-talk-name=org.freedesktop.DBus
+  - --talk-name=org.freedesktop.DBus
+
+  # Enable access to Flatpak host commands (not supported by Flathub's build system)
+  - --talk-name=org.freedesktop.Flatpak
+
+  # Enable access to Freedesktop Notifications
+  - --talk-name=org.freedesktop.Notifications
+
+  # Enable access to Freedesktop Secret Service & KWallet
+  - --talk-name=org.freedesktop.secrets
+  - --talk-name=org.kde.kwalletd6
+
+modules:
+  - name: zed
+    buildsystem: simple
+    build-commands:
+      # Install Zed
+      - install -Dm 755 bin/* --target-directory ${FLATPAK_DEST}/bin
+      - install -Dm 755 lib/* --target-directory ${FLATPAK_DEST}/lib
+      - install -Dm 755 libexec/* --target-directory ${FLATPAK_DEST}/libexec
+
+      # Install {Desktop,MetaInfo}
+      - install -Dm 644 share/applications/* --target-directory ${FLATPAK_DEST}/share/applications
+      - install -Dm 644 ${FLATPAK_ID}.metainfo.xml --target-directory ${FLATPAK_DEST}/share/metainfo
+
+      # Install Icons (Max. size 512x512)
+      - install -Dm 644 share/icons/hicolor/512x512/apps/zed.png --target-directory ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps
```

Comment:
Since you have the SVG, it would be better to install it in the correct location and the build will generate the icons for the supported resolutions automatically — and the next line could be simplified as well.

==> https://github.com/flathub/com.jetbrains.IntelliJ-IDEA-Ultimate/blob/master/com.jetbrains.IntelliJ-IDEA-Ultimate.yaml#L155

Reply:
The SVG they have isn't an actual desktop icon (no background, extends to edge of image, etc.), so it shouldn't be used. Not sure why they have it checked in.